.gitignore

```diff
@@ -12,3 +12,14 @@ Cargo.lock
 
 # MSVC Windows builds of rustc generate these, which store debugging information
 *.pdb
+
+# Circuits
+*.json
+out
+out.r1cs
+out.wtns
+witness
+proving.key
+verification.key
+*.sol
```

Comment:
This looks way too broad. We're here to learn so let's spend a moment discussing `.gitignore` in general:
- Note that these rules will apply anywhere in the repo. I.e. any file or directory called `out` anywhere, in any subdirectory, will not show up as a new file in `git status`.
    - If you use `out/` instead, it will only ignore directories.
    - If this is one specific dir at a known location, prefix it with a path. In case of something at repo root, that would be just a slash: `/out`. If it was inside, say `zkroller/circuits`, you'd use `/zkroller/circuits/out` instead.
- I would avoid broadly ignoring all files of one kind unless you know for sure that these files are always temporary and do not need to be committed. For example it's pretty safe to say you'd never commit `*.pdb` since that's just binary debug data and the compiler can recreate it for you. But I would not ignore all `*.json` files or all `*.sol` files. Even if Zokrates is generating these for you now, we will definitely have other files of this kind in the future that we will want to store in the repo.

See [`.gitignore` PATTERN FORMAT](https://git-scm.com/docs/gitignore#_pattern_format) for more details.

With that said, we probably want it to look like this:
```suggestion
/out/
/out.wtns
/witness
/proving.key
/verification.key
/verifier.sol
```

But even this seems a bit too fine grained. We'd be better off just ensuring that `zokrates` commands drop their files in one specific directory that we can easily remove.